fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java

```diff
@@ -633,28 +633,32 @@ public synchronized List<Long> seqChooseBackendIds(int backendNum, boolean needA
             lastBackendId = lastBackendIdForOther;
         }
 
-        // host -> BE list
-        Map<String, List<Backend>> backendMaps = Maps.newHashMap();
-        for (Backend backend : srcBackends) {
-            // If needAvailable is true, unavailable backend won't go into the pick list
-            if (needAvailable && !backend.isAvailable()) {
-                continue;
-            }
+        List<Backend> backends = Lists.newArrayList();
+        if (Config.enable_choose_backends_on_the_same_host) {
+            backends = srcBackends;
```

Comment:
should filter the dead backend when need available is true

Reply:
Old choose policy still not filter dead backend. I don't change this policy for this pr?

Reply:
I see that in the original code
```
                // If needAvailable is true, unavailable backend won't go into the pick list
                if (needAvailable && !backend.isAvailable()) {
                    continue;
                }
```

Reply:
fix it.